stlab/concurrency/future.hpp

```diff
@@ -396,7 +396,7 @@ struct shared_base<T, enable_if_copyable<T>> : std::enable_shared_from_this<shar
             if (_exception) std::rethrow_exception(_exception);
             return _result;
         }
-        return {};
```

Comment:
What is the reason behind this change? From my point of view returning the dafault c'tor is  perfectly fine and illustrates more clearly what is happening. Or?

Reply:
I agree with you... I had to solve a GCC10 warning

```
future.hpp:399:17: error: ‘<anonymous>’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  399 |         return {};
```
https://github.com/fpelliccioni/libraries/runs/1544232233?check_suite_focus=true#step:7:1022